tests/e2e/backup_restore_test.go

```diff
@@ -210,6 +210,56 @@ var _ = Describe("Backup and restore", Label(tests.LabelBackupRestore), func() {
 			})
 		})
 
+		// Test that the restore works if the source cluster has a custom
+		// backup.barmanObjectStore.serverName that is different than the cluster name
+		It("backs up and restores a cluster with custom backup serverName", func() {
+			const (
+				targetDBOne              = "test"
+				targetDBTwo              = "test1"
+				targetDBSecret           = "secret_test"
+				testTableName            = "test_table"
+				clusterRestoreSampleFile = fixturesDir + "/backup/cluster-from-restore-custom.yaml"
+				// clusterWithMinioCustomSampleFile has metadata.name != backup.barmanObjectStore.serverName
+				clusterWithMinioCustomSampleFile = fixturesDir + "/backup/minio/cluster-with-backup-minio-custom-servername.yaml"
+				backupFileCustom                 = fixturesDir + "/backup/minio/backup-minio-custom-servername.yaml"
+			)
+
+			clusterName, err := env.GetResourceNameFromYAML(clusterWithMinioCustomSampleFile)
```

Comment:
use "clusterName, err :=" here means initializing a new variable `clusterName` instead of assigning another value to the global variable `clusterName` so the failure of the test will not trigger saving the cluster environment